app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java

```diff
@@ -123,7 +126,11 @@ private void search(HybridFile directory, final SearchFilter filter) {
           activity.get(),
           rootMode,
           file -> {
-            if (!isCancelled()) {
+            boolean showHiddenFiles =
+                PreferenceManager.getDefaultSharedPreferences(activity.get())
+                    .getBoolean(PREFERENCE_SHOW_HIDDENFILES, false);
+
+            if (!isCancelled() && (showHiddenFiles || !(!showHiddenFiles && file.isHidden()))) {
```

Comment:
Condition can be improved
`(!isCancelled() && (showHiddenFiles || !file.isHidden()))`

Reply:
> Condition can be improved
> `(!isCancelled() && (showHiddenFiles || !file.isHidden()))`

✅ Done.. Please check...